From 6565c39cce97fef28b3bf4151b4799bdec422bb6 Mon Sep 17 00:00:00 2001
From: Peter Vary <pvary@cloudera.com>
Date: Fri, 26 Aug 2016 10:10:13 -0500
Subject: [PATCH 0741/1164] CDH-43555 : HIVE-14360: Starting BeeLine after
 using !save, there is an error logged: "Error
 setting configuration: conf"

Change-Id: I09e047a27aee6eb6acbd900d3ddcebd53f3531e5
---
 .../java/org/apache/hive/beeline/BeeLineOpts.java  |    1 +
 .../org/apache/hive/beeline/TestBeeLineOpts.java   |   41 ++++++++++++++++++++
 2 files changed, 42 insertions(+)
 create mode 100644 beeline/src/test/org/apache/hive/beeline/TestBeeLineOpts.java

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java b/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
index 3c86e41..e9254a8 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
@@ -604,6 +604,7 @@ public boolean isHelpAsked() {
     return helpAsked;
   }
 
+  @Ignore
   public HiveConf getConf() {
     return conf;
   }
diff --git a/beeline/src/test/org/apache/hive/beeline/TestBeeLineOpts.java b/beeline/src/test/org/apache/hive/beeline/TestBeeLineOpts.java
new file mode 100644
index 0000000..63d1885
--- /dev/null
+++ b/beeline/src/test/org/apache/hive/beeline/TestBeeLineOpts.java
@@ -0,0 +1,41 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hive.beeline;
+
+import junit.framework.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.io.*;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static org.mockito.Mockito.*;
+
+public class TestBeeLineOpts {
+  @Test
+  public void testPropertyNamesSet() throws Exception {
+    BeeLine mockBeeLine = mock(BeeLine.class);
+    when(mockBeeLine.isBeeLine()).thenReturn(true);
+    when(mockBeeLine.getReflector()).thenReturn(new Reflector(mockBeeLine));
+    BeeLineOpts beeLineOpts = new BeeLineOpts(mockBeeLine, System.getProperties());
+    Assert.assertFalse(beeLineOpts.propertyNamesSet().contains("conf"));
+  }
+}
-- 
1.7.9.5

